experimental/cache_associativity.cc

```diff
@@ -0,0 +1,85 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <string>
+
+// NOTE: the output does not make sense for now
+// TODO: give compilation instructions instead of including nonlocal files
```

Comment:
As discussed offline, can you please add a CMake target to build this file?

Reply:
Added as a new demo to the CMakeLists.txt